File: RentalMS/src/main/java/org/reservation/rentalms/model/vo/ReturnItem.java
package org.reservation.rentalms.model.vo;

import jakarta.persistence.Embeddable;
import jakarta.persistence.Embedded;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.reservation.rentalms.model.RentalItem;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Embeddable
public class ReturnItem {
    @Embedded
    private RentalItem rentalItem;
    private LocalDate returnDate;

    public static ReturnItem createReturnItem(RentalItem rentalItem) {
        return new ReturnItem(rentalItem, LocalDate.now());
    }

    public static ReturnItem sample() {
        return new ReturnItem(RentalItem.sample(), LocalDate.now());
    }
}

File: RentalMS_kotlin/src/main/kotlin/org/reservation/rentalms_kotlin/model/vo/ReturnItem.kt
package org.reservation.rentalms_kotlin.model.vo

import org.reservation.rentalms_kotlin.model.RentalItem
import java.time.LocalDate

data class ReturnItem(
        val rentalItem: RentalItem,
        val returnData: LocalDate?=null,
) {
    companion object {
        fun createReturnItem(rentalItem: RentalItem): ReturnItem {
            return ReturnItem(rentalItem, LocalDate.now());
        }

        fun sample(): ReturnItem {
            return ReturnItem(RentalItem.sample());
        }
    }
}
